src/cli/test/task-generate.spec.ts

```diff
@@ -0,0 +1,136 @@
+import type * as d from '../../declarations';
+import { taskGenerate, getBoilerplateByExtension, GenerableExtension, BoilerplateFile } from '../task-generate';
+import { mockConfig, mockStencilSystem } from '@stencil/core/testing';
+import * as utils from '../../utils/validation';
+
+import * as coreCompiler from '@stencil/core/compiler';
+import { CoreCompiler } from '../load-compiler';
+
+const promptMock = jest.fn().mockResolvedValue('my-component');
+
+jest.mock('prompts', () => ({
+  prompt: promptMock,
+}));
+
+const setup = async () => {
```

Comment:
One thing to call out here that I meant to do earlier (@rwaskiewicz) I sometimes like to write the setup / mocking / etc code for a test suite in a `setup` function like that rather than in `beforeEach`, just because I've gotten bitten so many times by some weirdness relating to `let`-ing a bunch of variables in the `describe` scope and then modifying them / not knowing where something is coming from, etc. I think this is a helpful pattern to just return from the function what you need to run the tests, but it's also not the standard pattern we're following in other spec files. So anyway, let me know if you think we should stick to the `beforeEach` pattern, and I can change it over if so

Reply:
I don't have a strong preference here - mocking/spying/stubbing in JavaScript is still "weird" to me compared to languages like Java 😆 - as long as we can understand what's going on in the suite, I think we can accept either.

Reply:
but JavaScript has Java right in the name...are you telling me they work differently??? :P